pkg/sql/catalog/randgen/util.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package randgen
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/dbdesc"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/nstree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/catid"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/errors"
+)
+
+// isAdmin returns true if the current user has the admin role.
+func (g *testSchemaGenerator) isAdmin(ctx context.Context) bool {
+	isAdmin, err := g.cat.HasAdminRole(ctx)
+	if err != nil {
+		panic(genError{err})
+	}
+	return isAdmin
+}
+
+// dbNames retrieves the list of database names in the schema.
+func (g *testSchemaGenerator) dbNames(ctx context.Context) map[string]struct{} {
+	dbs, err := g.coll.GetAllDatabases(ctx, g.txn)
+	if err != nil {
+		panic(genError{err})
+	}
+	res := make(map[string]struct{})
+	_ = dbs.ForEachNamespaceEntry(func(e nstree.NamespaceEntry) error {
+		res[e.GetName()] = struct{}{}
+		return nil
+	})
+	return res
+}
+
+// scNamesInDb retrieves the list of schema names in the given
+// database.
+func (g *testSchemaGenerator) scNamesInDb(
+	ctx context.Context, db catalog.DatabaseDescriptor,
+) map[string]struct{} {
+	res := make(map[string]struct{})
+	_ = db.ForEachSchema(func(_ descpb.ID, name string) error {
+		res[name] = struct{}{}
+		return nil
+	})
```

Comment:
Consider using `coll.GetAllSchemasInDatabase` here, unless you don't care that this won't work when temporary schemas are involved.